blob/driver/driver.go

```diff
@@ -61,24 +60,40 @@ type WriterOptions struct {
 	BufferSize int
 }
 
-// ObjectAttrs contains metadata of an object.
-type ObjectAttrs struct {
+// ReaderAttributes contains a subset of attributes about a blob that are
+// accessible from Reader. Use Bucket.Attributes to get the full set of
+// attributes.
+type ReaderAttributes struct {
+	// ContentType is the MIME type of the blob object. It must not be empty.
+	ContentType string
 	// Size is the size of the object in bytes.
 	Size int64
+}
+
+// Attributes contains attributes about a blob.
+type Attributes struct {
 	// ContentType is the MIME type of the blob object. It must not be empty.
 	ContentType string
+	// Size is the size of the object in bytes.
+	Size int64
+
 	// ModTime is the modified time of the blob object. Will be time.Time zero value if unknown.
 	ModTime time.Time
```

Comment:
Wasn't this already a reader attribute? Why the change? (Apologies if this was already addressed elsewhere.)

Reply:
Yes, it was a Reader attribute, and I could put it in ReaderAttributes. I didn't largely because GCS didn't originally support it; Tuo filed a feature request and they added it for us. I take this as a signal that it's an attribute that not all providers might support as part of reading.

I don't feel strongly about this either way though.

Reply:
I feel like it should be included, because otherwise we're discarding the information. We already document that it might not be filled in.

Reply:
Done.